Warnings (C5245) when having PUGIXML_NO_STL and PUGIXML_API defined

Hi,

When compiling pugixml and having the PUGIXML_API/PUGIXML_CLASS definitions set as well as PUGIXML_NO_STL, I get a bunch of C5245 warnings:

```
1>pugixml.cpp(262,22): warning C5245: 'pugi::impl::`anonymous-namespace'::strlength_wide': unreferenced function with internal linkage has been removed
1>pugixml.cpp(2311,22): warning C5245: 'pugi::impl::`anonymous-namespace'::as_utf8_begin': unreferenced function with internal linkage has been removed
1>pugixml.cpp(2317,20): warning C5245: 'pugi::impl::`anonymous-namespace'::as_utf8_end': unreferenced function with internal linkage has been removed
```

I've added ```#   pragma warning(disable: 5245) // unreferenced function with internal linkage has been removed``` to ln54 in the cpp to resolve this. Might be good to either ifdef the functions for ```#define PUGIXML_NO_STL``` or add the warning disable pragma.

--- a/src/pugixml.cpp
+++ b/src/pugixml.cpp
@@ -5054,6 +5054,14 @@ PUGI_IMPL_NS_BEGIN
 #if defined(PUGI_IMPL_MSVC_CRT_VERSION) || defined(__BORLANDC__) || (defined(__MINGW32__) && (!defined(__STRICT_ANSI__) || defined(__MINGW64_VERSION_MAJOR)))
 	PUGI_IMPL_FN FILE* open_file_wide(const wchar_t* path, const wchar_t* mode)
 	{
+#ifdef PUGIXML_NO_STL
+		// ensure these symbols are consistently referenced to avoid 'unreferenced function' warnings
+		// note that generally these functions are used in STL builds, but PUGIXML_NO_STL leaves the only usage in convert_path_heap
+		(void)&as_utf8_begin;
+		(void)&as_utf8_end;
+		(void)&strlength_wide;
+#endif
+
 #if defined(PUGI_IMPL_MSVC_CRT_VERSION) && PUGI_IMPL_MSVC_CRT_VERSION >= 1400
 		FILE* file = NULL;
 		return _wfopen_s(&file, path, mode) == 0 ? file : NULL;

